### PR TITLE
1342/multiple photos upload

### PR DESCRIPTION
**Pull request by jaredcwhite**
## Issue Overview

This PR addresses #1342 

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Updates the photo uploading functionality for Partners so that more than one image can be uploaded, the order of images can be adjusted, and existing images can be deleted.

## How Can This Be Tested/Reviewed?

Edit a listing on the Partners portal, for example: https://deploy-preview-1410--detroit-partners-dev.netlify.app/listings/f8e7639c-015b-4e3d-bcf2-6a9bd08b9ffa/edit

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

**Comment by jaredcwhite** [comment]
I still need to update the Cypress E2E test…

**Comment by jaredcwhite** [comment]
OK, looks like all checks are now passing!

**Comment by ColinBuyck** [comment]
I'm noticing an unexpected behavior when there are no photos uploaded. It shows a row for primary photo that can be deleted despite there being no photo uploaded.

https://user-images.githubusercontent.com/53269332/182241685-9b81b82c-dde7-4cf4-9256-8503a4e8eb25.mov

**Review by ColinBuyck — CHANGES_REQUESTED** [review]
Unexpected empty state

**Comment by jaredcwhite** [comment]
@ColinBuyck Thanks for catching this! I found some code in the main listings page that was literally creating a blank new image item if none existed. 😄 Removed that, and I think we're good!

**Review by ColinBuyck — CHANGES_REQUESTED** [review]
When selecting multiple photos to upload, there was no error shown and no photos successfully uploaded. Looking at the issue, it's unclear if batch upload is required but it seems it would be be very useful for this! What are your thoughts @jaredcwhite?

https://user-images.githubusercontent.com/53269332/182735572-1f471326-7bee-4447-80a8-a782f857b49d.mov

**Comment by jaredcwhite** [comment]
@ColinBuyck OK, I think I've resolved the weird issue where you could select multiple images and then it wouldn't work right. It'll only allow one upload at a time. We can figure out how to handle multiple concurrent uploads in a later pass.

**Review by emilyjablonski — CHANGES_REQUESTED** [review]
Just a few comments!
Visually:
* Missing the subtext under `Upload File` about cropping from the Figma
* Missing the `Photos` title above the table from the Figma
* I added two photos, and the directory is using the second photo while the detail is using the primary photo
* At the moment I can add what seems like unlimited photos which I don't think is intentional

**Comment by jaredcwhite** [comment]
Thanks for the review @emilyjablonski. I don't know the answer to a few questions so I'll need to defer to @slowbot…

> Missing the subtext under Upload File about cropping from the Figma
> Missing the Photos title above the table from the Figma
> At the moment I can add what seems like unlimited photos which I don't think is intentional

We're cropping ourselves automatically, so do we need to mention any of that in the help text? Can you review what's there currently vs. the Figma and provide guidance on the labeling? And do we need to limit the number of photos? 10? 15? 🤷🏻‍♂️

> I added two photos, and the directory is using the second photo while the detail is using the primary photo

I think any changes to the listings on the public site would be covered by #1343

**Comment by jaredcwhite** [comment]
During standup we discussed merging this in after the immediate code fixes, and addressing the UX issues in a followup task. Think we're OK to merge this now?

**Comment by jaredcwhite** [comment]
@slowbot How about this for temporary copy before we get to the batch upload feature:

**Select JPEG or PNG file to upload. Please upload horizontal images only at approximately 1440px. Up to 10 uploaded images allowed.**

**Comment by jaredcwhite** [comment]
@slowbot On the drawer buttons issue, it looks like the Detroit repo variant of the drawer is different than Bloom core now, due to a change back in April (related to the Filters UI): https://github.com/CityOfDetroit/bloom/commit/aafe1ec821efb2bbfada70bde901252dd265913d#diff-4432059db5561e8d10f2951e01b4c7db8bb51a3905da82e1a94036844d996ecd

The action buttons are pushed to the bottom of the window, but not on core.

How shall we handle this?

**Comment by jaredcwhite** [comment]
@slowbot That seems reasonable. I can make the change in Detroit and open a backport PR in core.

**Comment by jaredcwhite** [comment]
I realized the solution in some other parts of the Detroit UI was that the buttons are just within the drawer group and aren't provided via the actions prop. So…I think I'll just do that for now, and we can include this in our audit of UI component variance and what needs to be normalized across repos.

**Comment by jaredcwhite** [comment]
OK, this is ready for a final review.

**Review by ColinBuyck — CHANGES_REQUESTED** [review]
I am noticing some styling discrepancies between the view below and the Figma:
Screen Shot 2022-08-17 at 4 19 44 PM

1. The "Photos" sub-header is misaligned with the Sort column header but aligned in the Figma
2. The Figma shows unique font settings for the "Upload File" text and the "Select JPEG or PNG..." but those don't seem reflected here

**Comment by jaredcwhite** [comment]
I addressed your feedback @ColinBuyck. I think having the flush left sortable column makes sense across the board, but I didn't notice any examples of that in Detroit. Maybe we'll want to revisit that in core for the preferences and programs. Anyway…

I wasn't sure exactly how to handle the image limit, so I just added this little notice and removed the dropzone component when it's at 10. Maybe we'd need a signoff first from @slowbot?

image

File: sites/partners/src/listings/PaperListingForm/sections/ListingPhotos.tsx
Comment (ColinBuyck) [inline sites/partners/src/listings/PaperListingForm/sections/ListingPhotos.tsx:284]: UI changes look good! I think this just needs to be updated to 10 since it currently limits it to 3 images.

**Review by ColinBuyck — CHANGES_REQUESTED** [review]

**Review by ColinBuyck — APPROVED** [review]
LGTM!

File: sites/partners/cypress/integration/listing.spec.ts
Comment (emilyjablonski) [inline sites/partners/cypress/integration/listing.spec.ts:105]: If we're not going to write a test here do you mind adding a ticket to do so?

Comment (jaredcwhite) [reply sites/partners/cypress/integration/listing.spec.ts:105]: @emilyjablonski Sure, I'll see if I can expand the test there a bit.

File: sites/partners/cypress/integration/listing.spec.ts
Comment (emilyjablonski) [inline sites/partners/cypress/integration/listing.spec.ts:122]: can you use a similar naming scheme as the existing image with a uuid? the intention is to be able to search by that image name and have it be unique in cloudinary to do cleanup

**Review by emilyjablonski — APPROVED** [review]
LGTM after just renaming the new photo, thank you for the new test!

File: sites/partners/cypress/integration/listing.spec.ts
Comment (jaredcwhite) [reply sites/partners/cypress/integration/listing.spec.ts:122]: Oh sure, I didn't realize why it had the uuid and thought that was because of some other automated step.

**Pull request merged**
